src/Data/Unfoldable.purs

```diff
@@ -38,3 +39,58 @@ instance unfoldableArray :: Unfoldable Array where
           writeSTRef seed b2
           return false
     return arr))
+
+-- | Replicate a value some natural number of times.
+-- | For example:
+-- |
+-- | ~~~ purescript
+-- | replicate 2 "foo" == ["foo", "foo"] :: Array String
+-- | ~~~
+replicate :: forall f a. (Unfoldable f) => Int -> a -> f a
+replicate n v = unfoldr step n
+  where
+    step :: Int -> Maybe (Tuple a Int)
+    step i =
+      if i <= 0 then Nothing
+      else Just (Tuple v (i - 1))
+
+-- | Perform an Applicative action `n` times, and accumulate all the results.
+replicateA :: forall m f a. (Applicative m, Unfoldable f, Traversable f) =>
+              Int -> m a -> m (f a)
+replicateA n m = sequence (replicate n m)
+
+-- | The empty container - unfolded with zero iterations.
+-- | For example:
+-- |
+-- | ~~~ purescript
+-- | empty == [] :: forall a. Array a
+-- | ~~~
+empty :: forall f a. (Unfoldable f) => f a
```

Comment:
I'd rather avoid having naming collisions between two core control libraries if possible, unless one is a generalization of the other. Since I'm not sure if this is a generalization of `Control.Plus.empty` in general, could we rename it, please? `nil`, or `none` perhaps?